### PR TITLE
st2 [1847][IMP] purchase_report_custom_hls: Delete Shipping Address

### PR DESCRIPTION
**Pull request by TomohiroAbe-Quartile**
[1847](https://www.quartile.co/web#id=1847&action=771&model=project.task&view_type=form&menu_id=505)

t-else or t-set tag in purchase_stock.report_purchaseorder_document, didn't work invisible attribute.
Maybe views relationship are complex.
Add style-tag in div or other html tag, it works.

### CODE REVIEW

File: purchase_report_custom_hls/report/purchase_order_report_templates.xml [hunk 1]
```diff
@@ -32,4 +35,11 @@
             </div>
         </xpath>
     </template>
+    <template
+        id="report_purchaseorder_document"
+        inherit_id="purchase_stock.report_purchaseorder_document"
+    >
+        <xpath expr="//t[@t-else='']" position="replace">
+        </xpath>
```
Comment (TomohiroAbe-Quartile) [inline purchase_report_custom_hls/report/purchase_order_report_templates.xml]: It doesn't workthat use invisible attribute.

File: purchase_report_custom_hls/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -1,5 +1,6 @@
 This module does the following:
 
 - Shows the payment terms and the signature image of the representative on the purchase order.
+- Remove shipping address from purchase order.
```
Comment (yostashiro) [inline purchase_report_custom_hls/readme/DESCRIPTION.rst]:
```suggestion
- Removes shipping address from the printed purchase order.
```

File: purchase_report_custom_hls/report/purchase_order_report_templates.xml [hunk 3]
```diff
@@ -11,6 +11,9 @@
                 <span t-field="o.payment_term_id"/>
             </div>
         </xpath>
+        <xpath expr="//t[@t-if='o.dest_address_id']" position="attributes">
```
Comment (yostashiro) [inline purchase_report_custom_hls/report/purchase_order_report_templates.xml]:
```suggestion
        <xpath expr="//t[@t-set='information_block']" position="attributes">
```

File: purchase_report_custom_hls/report/purchase_order_report_templates.xml (hunk 1 above)
Comment (yostashiro) [inline purchase_report_custom_hls/report/purchase_order_report_templates.xml]:
```suggestion
        <xpath expr="//t[@t-set='information_block']" position="attributes">
            <attribute name="invisible">1</attribute>
        </xpath>
```

**Review by yostashiro — COMMENTED** [review]
@TomohiroAbe-Quartile Can you please check if the suggested changes work nicely?

The reasons of the suggestions:
- We want to avoid using `replace` where we can.  `replace` adds the risk of breaking the system when another custom module tries to use an element in there.
- We want to apply the changes in the same style where we can.

File: purchase_report_custom_hls/report/purchase_order_report_templates.xml (hunk 1 above)
Comment (TomohiroAbe-Quartile) [reply purchase_report_custom_hls/report/purchase_order_report_templates.xml]: I tried but it doesn't worked.

File: purchase_report_custom_hls/report/purchase_order_report_templates.xml (hunk 3 above)
Comment (TomohiroAbe-Quartile) [reply purchase_report_custom_hls/report/purchase_order_report_templates.xml]: Thank you, I edited it.

File: purchase_report_custom_hls/readme/DESCRIPTION.rst (hunk 2 above)
Comment (TomohiroAbe-Quartile) [reply purchase_report_custom_hls/readme/DESCRIPTION.rst]: Thanks, I regenerated readme

File: purchase_report_custom_hls/report/purchase_order_report_templates.xml [hunk 4]
```diff
@@ -32,4 +35,16 @@
...
+    <template
+        id="report_purchaseorder_document"
+        inherit_id="purchase_stock.report_purchaseorder_document"
+    >
+        <xpath expr="//t[@t-else='']/t[@t-set='information_block']/strong" position="attributes">
+            <attribute name="style">display: none;</attribute>
+        </xpath>
+        <xpath expr="//div[@t-if='o.picking_type_id and o.picking_type_id.warehouse_id']" position="attributes">
+            <attribute name="style">display: none;</attribute>
+        </xpath>
```
Comment (TomohiroAbe-Quartile) [inline purchase_report_custom_hls/report/purchase_order_report_templates.xml]: @yostashiro 
Edit this line just add HTML style attribute.
It doesn't work "invisible" attr in t-set tag.

Comment (yostashiro) [inline purchase_report_custom_hls/report/purchase_order_report_templates.xml]: I don't feel comfortable with this approach. If this and https://github.com/qrtl/hls-custom/pull/156/files#diff-2466a289dc9319e6af09eff092e75e482bf4775151dd4251c3f7db981849e854R7-R9 work, why not below?

```
        <xpath expr="//t[@t-else='']/t[@t-set='information_block']" position="attributes">
            <attribute name="invisible">1</attribute>
        </xpath>
```

Comment (TomohiroAbe-Quartile) [reply purchase_report_custom_hls/report/purchase_order_report_templates.xml]: Sorry, I couldn't found.
I debug purchase_stock's models and add "t-att-style" in t-if or t-set tag. It doesn't works.

I want to merge this PR. After then, I will investigate why t-if or t-set aren't accepted any attribute.

**Comment by yostashiro** [comment]
@TomohiroAbe-Quartile Can you please take care of the rest of the steps:
- Re-generate README
- Add translation

I ended up using the groups attribute.  Although I am still not convinced that this is the best solution, it is now slightly more configurable.

File: purchase_report_custom_hls/ja_JP (3).po [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# Translation of Odoo Server.
```
Comment (yostashiro) [inline purchase_report_custom_hls/ja_JP (3).po]: Remove this file.

File: purchase_report_custom_hls/i18n/ja_JP.po [hunk 6]
```diff
@@ -16,7 +16,22 @@ msgstr ""
...
-msgid "Signature"
-msgstr "署名"
+#: model:ir.model,name:purchase_report_custom_hls.model_res_config_settings
+msgid "Config Settings"
+msgstr "コンフィグ設定"
+
+#. module: purchase_report_custom_hls
+#: model:ir.model.fields,field_description:purchase_report_custom_hls.field_res_config_settings__group_show_purchase_shipping_address
+msgid "Show Purchase Shipping Address"
+msgstr "購買オーダへの配送先住所の表示"
```
Comment (yostashiro) [inline purchase_report_custom_hls/i18n/ja_JP.po]:
```suggestion
msgstr "購買出荷先アドレス表示"
```

File: purchase_report_custom_hls/i18n/ja_JP.po [hunk 7]
```diff
@@ -16,7 +16,22 @@ msgstr ""
...
+
+#. module: purchase_report_custom_hls
+#: model:res.groups,name:purchase_report_custom_hls.group_show_purchase_shipping_address
+msgid "Show Shipping Address on Purchase Order"
+msgstr "購買オーダ配送先住所"
```
Comment (yostashiro) [inline purchase_report_custom_hls/i18n/ja_JP.po]:
```suggestion
msgstr "購買オーダに出荷先アドレスを表示"
```

File: purchase_report_custom_hls/i18n/ja_JP.po [hunk 8]
```diff
@@ -16,7 +16,22 @@ msgstr ""
...
+
+#. module: purchase_report_custom_hls
+#: model_terms:ir.ui.view,arch_db:purchase_report_custom_hls.res_config_settings_view_form_purchase
+msgid "Show the shipping address on the printed purchase order"
+msgstr "購買オーダに出荷宛先を表示する"
```
Comment (yostashiro) [inline purchase_report_custom_hls/i18n/ja_JP.po]:
```suggestion
msgstr "購買オーダ印刷で出荷先アドレスを表示する"
```

**Review by yostashiro — APPROVED** [review]
Please correct the points (at least be consistent with the translation of 'shipping address') that I raise, and go ahead and merge this.

**Pull request merged**